bindings/muen/muen-net.c

```diff
@@ -103,62 +145,80 @@ void generate_mac_addr(uint8_t *addr)
     addr[0] |= 0x02;
 }
 
-void net_init(void)
+static bool muen_net_dev_init(const char *name, struct muen_net_device *device,
+        struct solo5_net_info *info)
 {
-    char mac_str[18];
     const uint64_t epoch = muen_get_sched_start();
+    char mac_str[18];
+    char buffer[64];
+    snprintf(buffer, sizeof buffer, "%s|out", name);
     const struct muen_resource_type *const
-        chan_out = muen_get_resource("net_out", MUEN_RES_MEMORY);
+        chan_out = muen_get_resource(buffer, MUEN_RES_MEMORY);
+	snprintf(buffer, sizeof buffer, "%s|in", name);
     const struct muen_resource_type *const
-        chan_in = muen_get_resource("net_in", MUEN_RES_MEMORY);
+        chan_in = muen_get_resource(buffer, MUEN_RES_MEMORY);
 
     if (!chan_out) {
-        log(WARN, "Solo5: Net: No output channel\n");
-        return;
+        log(WARN, "Solo5: Net: '%s': No output channel\n", name);
+        return false;
```

Comment:
Is it possible to end up in a state where `chan_out` is `NULL` but `chan_in` is allocated/leaked here?

Reply:
Yes, that is possible however I did not think it is too problematic since `acquired` will only be set if everything succeeds.
I can move all of the assignments to the end of the function after all checks for both channels have passed.